### DISCUSSION TITLE
Doris 3.0.1 Storage Vault对接Azure Blob报UnknownHostException

### DISCUSSION THREAD

**Discussion by cjj198909** [apache/doris#40700] [A - General / Q&A]
创建Azure Blob的Storage Vault时报错UnknownHostException，在fe/be/ms/re中均可以正常解析blob endpoint。
SQL：
CREATE STORAGE VAULT IF NOT EXISTS ssb_azure_vault2
    PROPERTIES (
        "type"="S3",
        "s3.endpoint" = "studysa2024.blob.core.windows.net/",
        "s3.bucket" = "studysa2024",
        "s3.root.path" = "doris-cloud",
        "s3.access_key" = "studysa2024",
        "s3.secret_key" = "keyxxxx",
        "provider" = "AZURE"
    );

response:
ERROR 1105 (HY000): errCode = 2, detailMessage = ping s3 failed(upload), status: [COMMON_ERROR, msg: connect to s3 failed: Received an UnknownHostException when attempting to interact with a service. See cause for the exact endpoint that is failing to resolve. If this is happening on an endpoint that previously worked, there may be a network connectivity issue or your DNS cache could be storing endpoints for too long.], properties: "s3.region" = "blob", "s3.endpoint" = "http://studysa2024.blob.core.windows.net/", "provider" =

-----------

另外还想咨询下s3.bucket是否配置为Blob Storage Account Name，s3.root.path配置为Container以及prefix？

**Comment by ixzc** [comment]
建议英文提问，如使用中文提问，可以去ask.selectdb.com中文技术论坛。

**Comment by Nj-kol** [comment]
@cjj198909 Hey were you able to resolve this. I am facing a similar issue with Azure and Doris compute storage seperation

**Comment by cjj198909** [comment]
Hi Nilanjan. You can read this note I've compiled(
https://yembrek2c0.feishu.cn/docx/ULT8dYXKtoP1MrxSeNwcEYgVn0d?from=from_copylink).
The note is in Chinese and may need to be translated.

Nilanjan Sarkar ***@***.***> 于2025年11月11日周二 20:57写道：

**Reply by Nj-kol** [reply]
Hi @cjj198909, thank you so much for your help. This worked for Doris 3.0.x, but I think there are some issues with Doris 4.x. 

One follow-up question - In your experience, what storage tier did you use or recommend for Azure Blob Storage?

Also, if you can suggest some learnings or best practices from your experience of running Doris in compute-storage seperation mode with Azure as a Vault, that would really help me :)
